### PR TITLE
Refactor normalization

### PR DESCRIPTION
**Pull request by Kevin2**
enable batch-wise normalization
enable normalization for test data
add normalization test

### CODE REVIEW

File: tests/test_mnist_normalization.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+        assert not np.isinf(x_train).any()
+        assert not np.isinf(x_valid).any()
+
+        x_train = normalize(x_train, train_mean, train_std)
+        x_valid = normalize(x_valid, train_mean, train_std)
+
+        assert np.isclose(x_train.mean(), 0, atol=1e-1)
+        assert np.isclose(x_train.std(), 1, atol=1e-1)
+        assert np.isclose(x_valid.mean(), 0, atol=1e-1)
+        assert np.isclose(x_valid.std(), 1, atol=1e-1)
```
Comment (FritzGeise) [inline tests/test_mnist_normalization.py]: Check tolerance

File: dl_framework/callbacks.py [hunk 2]
```diff
@@ -97,78 +126,120 @@ def plot(self, skip_last=0):
...
+        valid_mean = torch.tensor(norm['valid_mean'].values[0]).float()
+        if self.in_train:
+            x[torch.isinf(x)] = train_mean
+            x = normalize(x, train_mean, train_std)
+            assert not torch.isinf(x).any()
+        if not self.in_train:
+            x[torch.isinf(x)] = valid_mean
+            x = normalize(x, train_mean, train_std)
+            assert not torch.isinf(x).any()
+        return x
```
[the remark this answers is not in the document]
Comment (FritzGeise) [reply dl_framework/callbacks.py]: At the given time not possible to replace

File: tests/test_mnist_normalization.py (hunk 1 above)
Comment (FritzGeise) [reply tests/test_mnist_normalization.py]: Validation Datensatz wird mit dem Mean und der Standardabweichung des Trainingsdatensatzes initialisiert und ist deswegen immer etwas ungenauer.
